clients/orderbook/src/worker.rs

```diff
@@ -1575,13 +1572,23 @@ where
 				_ = gossip_engine => {
 					error!(target: "orderbook", "📒 Gossip engine has terminated.");
 					return;
-				}
+				},
 				finality = finality_stream.next() => {
 					if let Some(finality) = finality {
+						info!(target: "orderbook", "📒 Waiting for orderbook pallet to become available...");
+						let at = BlockId::hash(finality.header.hash());
+						if self.runtime.runtime_api().validator_set(&at).ok().is_some() {
+							// we are good to go
+							self.operational.store(true, Ordering::Relaxed);
+						} else {
+							debug!(target: "orderbook", "📒 Waiting for orderbook pallet to become available...");
+							// dont process further
+							continue
+						}
 						if let Err(err) = self.handle_finality_notification(&finality).await {
 							error!(target: "orderbook", "📒 Error during finalized block import{}", err);
 						}
-					}else {
+					}	else {
```

Comment:
```suggestion
					} else {
```

Code style. Extra space should be removed.